packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -335,10 +338,34 @@ exports.createContentTypeNodes = ({
       // Get localized fields.
       const entryItemFields = _.mapValues(entryItem.fields, (v, k) => {
         const fieldProps = contentTypeItem.fields.find(field => field.id === k)
-        if (fieldProps.localized) {
-          return getField(v)
+
+        const localizedField = fieldProps.localized
+          ? getField(v)
+          : v[defaultLocale]
+
+        if (
+          fieldProps.type === `RichText` &&
+          richTextOptions.resolveFieldLocales
+        ) {
+          const contentTypesById = contentTypeItems.reduce(
```

Comment:
This effectively caches the content types by ID, so that the content types don't need to be searched for in the array for every entry. (This addresses [this comment](https://github.com/gatsbyjs/gatsby/pull/14616#discussion_r354259456) from @pvdz.)